docs(external): improve debugging guide (#22735)
vectordotdev/vector: website/content/en/guides/developer/debugging.md

2025-02-12T19:10:54.562556Z  WARN sink{component_kind="sink" component_id=sink_0 component_type=http}:request{request_id=19}: vector::sinks::util::retries: Retrying after response. reason=too many requests internal_log_rate_limit=true
```

{{< warning >}}

> The `component_sent_events_total` metrics for `sinks_0` is not increasing anymore.
> And also, we can now see a `429` code in the response status.

Reviewer: ```suggestion
The `component_sent_events_total` metric for `sinks_0` has stopped increasing, and we are observing a 429 response status code.
```

#### Scenario 3 - Smaller Batching

For this scenario, we will introduce a new limitation imposed by the server.

{{< warning >}}

> Our downstream server has a strict limit on the maximum payload size. Payloads larger than 8192 bytes will be rejected.

Reviewer: ```suggestion
Our downstream server enforces a strict limit on the maximum payload size. Payloads larger than 8192 bytes will be rejected.
```

{{< /warning >}}

Also, assume Vector sets the maximum batch size to 10MB.

{{< info >}}

> Always refer to the docs as the source of truth for this value.

Reviewer: ```suggestion
Always refer to the documentation as the source of truth for this value.
```
